LanguageFeatures/Private-fields-promotion/promotion_A02_t01.dart

```diff
@@ -12,7 +12,7 @@
 /// - There is no implicit noSuchMethod forwarder with the same name elsewhere
 ///   in the library.
 ///
-/// @description Checks that an instance field is promotable even if it is late
+/// @description Checks that an instance field is not promotable if it is late
```

Comment:
Nit: late final instance fields _are_ promotable, but they can't be promoted by _writing_ to them. They can only be promoted by reading from them and checking the type of the value that was read (just like final instance fields that aren't late).

So for example, this works fine:
```dart
class C {
  late final int? _i;
}
main() {
  C c = C();
  c._i = 1;
  if (c._i != null) {
    c._i.isOdd; // OK
  }
}
```

Reply:
I think LanguageFeatures/Private-fields-promotion/promotion_A02_t01.dart is a bit out of sync with this consideration: It doesn't actually test that a promotion occurs _based_ on a write operation, it just performs a write and doesn't rely on any promotions at that point. 

The next step in that test is to perform an `is` test, and I think that should give rise to a promotion.

So we should probably not have a `@description` that says "not promotable" and that's it.

Maybe we could use something like 'Checks that an instance field is promotable even if it is late (but it is promoted because of the type test, not because of the assignment)'.

Reply:
I think I wrote the previous comment on a web page where your previous comment hadn't been fetched. ;-)
To tests sounds good!